### PR TITLE
Updated arch sync script

### PR DESCRIPTION
**Pull request by BernardZhao**
Completes #507, mainly adapting the code from [here](https://git.server-speed.net/users/flo/bin/tree/syncrepo.sh). Has not yet been tested.

### CODE REVIEW

File: modules/ocf_mirrors/files/project/archlinux/sync-archive [hunk 1]
```diff
@@ -3,5 +3,45 @@
...
+
+rsync_cmd \
+    --exclude='*.links.tar.gz*' \
+    --exclude='/other' \
+    --exclude='/sources' \
+    --exclude='/iso' \
+    "${source_url}" \
+    "${target}"
+
+#echo "Last sync was $(date -d @$(cat ${target}/lastsync))"
```
Comment (dkess) [inline modules/ocf_mirrors/files/project/archlinux/sync-archive]: delete this last line, it's not needed

**Comment by abizer** [comment]
this is currently in prod on my environment in falling rocks, we're testing
it

**Comment by dkess** [comment]
Just took it off your environment. The script fails with this error:

```
/opt/mirrors/project/archlinux/sync-archive: 20: /opt/mirrors/project/archlinux/sync-archive: Syntax error: "(" unexpected (expecting "}")
```

**Comment by abizer** [comment]
wait you shouldn't have changed fallingrocks from my environment, it's running things for the migration to dev-fallingrocks, I just checked out Bernard's script

**Comment by dkess** [comment]
ok i canceled it, don't think anything actually ended up being changed, but can you please change it back until this gets fixed

**Comment by abizer** [comment]
done

@BernardZhao try running the script manually to fix the bugs and then push
a new version and we'll try that

**Comment by dkess** [comment]
lgtm, can you run this in prod for a bit before it gets merged for real?

**Comment by MCer4294967296** [comment]
Why are they using this `local -a cmd` array, as opposed to construct a string and `exec theString`? The latter one seems to work with POSIX shell.

**Comment by BernardZhao** [comment]
>Why are they using this `local -a cmd` array, as opposed to construct a string and `exec theString`? The latter one seems to work with POSIX shell.

@MCer4294967296 This was taken from the original script mentioned in #507. Making the script portable is an improvement I can make next

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kpengboy** [comment]
@MCer4294967296 probably because shell command strings are trickier to construct correctly, especially if you have variables that might have spaces or other weird characters in them. Constructing a command array generally leads to fewer nasty surprises.

**Comment by MCer4294967296** [comment]
guess we might take some risks to get a portable script

**Comment by dkess** [comment]
See [the OCF's current position on `bash` vs `sh`](https://github.com/ocf/utils#shellbash). If it's significant effort to use `sh`, we shouldn't bother, since we have bash everywhere anyways.

**Comment by BernardZhao** [comment]
AFAIK this should be working and has been tested, can it be merged?

**Comment by dkess** [comment]
Yup, you can merge it now.

**Comment by redplanks** [comment]
@BernardZhao The build failed because of a linting issue. When you're developing for our repos, please make sure to install pre-commit next time.

**Comment by cg505** [comment]
Also, why did we merge without checking Jenkins?

**Comment by BernardZhao** [comment]
@cg505 Foolishness on my part. @redplanks got it

**Comment by dkess** [comment]
my bad re: not checking Jenkins

**Comment by cg505** [comment]
no worries

**Pull request merged**